### DISCUSSION TITLE
raw.githubusercontent.com

### DISCUSSION THREAD

**Discussion by Complx21** [community/community#140054] [Programming Help]
### Select Topic Area

Question

### Body

I'm trying to connect my site to a GitHub repository and run an index.html page in an iframe using raw.githubusercontent.com, but I keep getting an error saying "raw.githubusercontent.com refused to connect".  If anyone could help that would be great.

**Comment by locflamedia** [comment]
The issue you're facing may be due to `X-Frame-Options` being set to `deny`, which causes GitHub to refuse the connection.

Here are some solutions you can try:

Use `gh-pages` instead of `raw.githubusercontent.com`.
Set up a proxy server to fetch content from `raw.githubusercontent.com` — although this can be quite complex.
Use other services like `Vercel`, `Netlify`, etc.

**Reply by Complx21** [reply]
what would I do with netlify becuz I'm already using it for deploying.

**Reply by R3posit** [reply]
Hmm really good idea.

**Comment by ThisIs-Developer** [comment]
hey @Complx21,

you’re running into the X-Frame-Options issue with GitHub. since you’re using Netlify, try this:

1. **host your HTML on Netlify**: deploy your index.html directly to your Netlify site instead of using raw.githubusercontent.com.

this should resolve the issue!

---
if it helps, feel free to mark it as the answer! 😪

**Reply by Complx21** [reply]
I'm using Netlify to host the website which is a click and play games website and the games are being pulled from a GitHub repo, there's a file called list.json and my code is working to pull the title and thumbnail,but I can 't get the game in the iframe. I'm hosting the website on Netlify and it wont let me deploy a second one, is there any other way to do it, if there is please tell me.

**Reply by ThisIs-Developer** [reply]
1. **github pages**: host your games directly on github pages and link to them in your iframe.

2. **cors configuration**: ensure the resources have the appropriate cors settings, but github may restrict this.

3. **proxy server**: set up a proxy server to fetch game content from github and serve it to your netlify site.

4. **static json data**: load game data as json and use javascript to render the games without iframes.

5. **embed game url**: use direct urls for your games if available instead of github raw links.

let me know if you need more details on any option!

**Reply by Complx21** [reply]
I could link the repo to a Netlify site but then how would I get the thumbnail and the title data, because the method I'm using it creates a button automatically for every item in the list then gets the data from the root file listed, like this:
`const listUrl = 'https://raw.githubusercontent.com/Complx21/files/main/list.json';

fetch(listUrl)
    .then(response => response.json())
    .then(data => {
        console.log(data); // Log the fetched data
        const gameButtons = document.getElementById('game-buttons');
        data.forEach(game => {
            const button = document.createElement('button');
            button.className = 'game-card';
            button.innerHTML = `
                <img src="https://raw.githubusercontent.com/Complx21/files/main/Icons/${game.gameroot.replace(/\//g, '')}.webp" class="button-img">
                <a>${game.game}</a>
            `;
            button.onclick = () => loadGame(`https://raw.githubusercontent.com/Complx21/files/main/${game.gameroot}index.html`);
            gameButtons.appendChild(button);
        });
    })
    .catch(error => console.error('Error fetching the game list:', error));

function loadGame(url) {
    const gameContainer = document.getElementById('game-container');
    const gameIframe = document.getElementById('game-iframe');
    gameIframe.src = url;
    gameContainer.style.display = 'block';
}

function closeGame() {
    const gameContainer = document.getElementById('game-container');
    const gameIframe = document.getElementById('game-iframe');
    gameIframe.src = '';
    gameContainer.style.display = 'none';
}`

**Reply by ThisIs-Developer** [reply]
### Adjust the Fetch URL
You should not need to change the existing `fetch` logic unless you change the hosting method of your `list.json`. Here’s your existing code slightly modified for clarity:

```javascript
const listUrl = 'https://raw.githubusercontent.com/Complx21/files/main/list.json';

fetch(listUrl)
  .then(response => response.json())
  .then(data => {
    const gameButtons = document.getElementById('game-buttons');
    data.forEach(game => {
      const button = document.createElement('button');
      button.className = 'game-card';
      button.innerHTML = `<img src="https://raw.githubusercontent.com/Complx21/files/main/Icons/${game.gameroot.replace(/\//g, '')}.webp" class="button-img"> <a>${game.game}</a>`;
      button.onclick = () => loadGame(`https://raw.githubusercontent.com/Complx21/files/main/${game.gameroot}index.html`);
      gameButtons.appendChild(button);
    });
  })
  .catch(error => console.error('Error fetching the game list:', error));

function loadGame(url) {
  const gameContainer = document.getElementById('game-container');
  const gameIframe = document.getElementById('game-iframe');
  gameIframe.src = url;
  gameContainer.style.display = 'block';
}

function closeGame() {
  const gameContainer = document.getElementById('game-container');
  const gameIframe = document.getElementById('game-iframe');
  gameIframe.src = '';
  gameContainer.style.display = 'none';
}
```

This way, you'll be able to maintain your current functionality while hosting on Netlify! Let me know if you need further assistance!

**Reply by R3posit** [reply]
Ow its looking very hard!

**Reply by ThisIs-Developer** [reply]
Thanks for your response! 

- **Iframe Loading**: If the URLs are correct and there are no CORS restrictions, the iframes should work.
- **Example JSON Structure**: 
  ```json
  [
    {
      "game": "Game Title 1",
      "gameroot": "path/to/game1/",
      "thumbnail": "path/to/thumbnail1.webp"
    }
  ]
  ```

#### If it doesn’t work, consider hosting the games on GitHub Pages, Vercel, or Cloudflare as alternatives.

**Reply by ThisIs-Developer** [reply]
Setting up a proxy server to fetch content from raw.githubusercontent.com is another option, but it's more complex and may involve security considerations.

**Comment by Complx21** [comment]
Thank you, but what did you change. Also this updated version I should be able to replace my code with this, then host the games repo (Which is separate from the site repo, btw) and the ifames will work?

**Reply by Complx21** [reply]
@ThisIs-Developer

**Reply by ThisIs-Developer** [reply]
@Complx21 😕 sorry, Hope it gets sorted soon!

**Reply by Complx21** [reply]
wdym?

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @Complx21 ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by joselopeztelus** [comment]
minishift 1.34.3 windows version is impacted with this problem.

**Reply by Complx21** [reply]
wdym???

**Comment by mecodeatlas** [comment]
@Complx21 we're going through our unanswered discussions, I know it has been a minute, but I wanted to check and see if you were able to get this solved?
